src/popper/runner_host.py

```diff
@@ -293,6 +293,198 @@ def _update_with_engine_config(self, container_args):
                 container_args[k] = update_with[k]
 
 
+class PodmanRunner(StepRunner):
+    """Runs steps in podman on the local machine."""
+
+    def __init__(self, init_podman_client=True, **kw):
+        super(PodmanRunner, self).__init__(**kw)
+
+        self._spawned_containers = set()
+        _, _, self._p_info = HostRunner._exec_cmd(["podman", "info"], logging=False)
+
+        if not init_podman_client:
+            return
+
+        try:
+            self._p_version = HostRunner._exec_cmd(["podman", "version"], logging=False)
+        except Exception as e:
+            log.debug(f"Podman error: {e}")
+            log.fail(f"Unable to connect to podman, is it installed?")
+
+        log.debug(f"Podman info: {pu.prettystr(self._p_info)}")
+
+    def run(self, step):
+        """Executes the given step in podman."""
+        cid = pu.sanitized_name(step.id, self._config.wid)
+        container = self._find_container(cid)
+
+        if not container and self._config.reuse:
+            log.fail(
+                f"Cannot find an existing container for step '{step.id}' to be reused"
+            )
+
+        if container and not self._config.reuse and not self._config.dry_run:
+            cmd = ["podman", "rm", "-f", container]
+            HostRunner._exec_cmd(cmd, logging=False)
+            container = None
+
+        if not container and not self._config.reuse:
+            container = self._create_container(cid, step)
+
+        log.info(f"[{step.id}] podman start")
+
+        if self._config.dry_run:
+            return 0
+
+        self._spawned_containers.add(container)
+
+        try:
+            cmd = ["podman", "start", container]
+            _, e, _, = HostRunner._exec_cmd(cmd, logging=False)
+
+        except Exception as exc:
+            log.fail(exc)
+
+        return e
+
+    def stop_running_tasks(self):
+        """Stop containers started by Popper."""
+        for c in self._spawned_containers:
+            log.info(f"Stopping container {c}")
+            _, ecode, _ = HostRunner._exec_cmd(["podman", "stop", c], logging=False)
+            if ecode != 0:
+                log.warning(f"Failed to stop the {c} container")
+
+    def _find_container(self, cid):
+        """Checks whether the container exists."""
+        cmd = ["podman", "inspect", "-f", str("{{.Id}}"), cid]
+        _, ecode, containers = HostRunner._exec_cmd(cmd, logging=False)
+
+        if ecode != 0:
```

Comment:
this needs to check whether the error is the one being expected, otherwise an actual error would be swallowed

Reply:
When you say error being expected what do you mean **Error: error inspecting object: no such object:** right? So how do you advise I represent this? I implemented that based on what I saw [here](https://github.com/getpopper/popper/blob/6831e0131eb48ce93461457a5b309d388595596a/src/popper/runner_host.py#L273). It's more of an implementation of DockerRunenr and there is nothing to do a check in DockerRunner

Reply:
Maybe I should log it? This is exactly what I did though

Reply:
yeah, checking that the returned output contains the `no such object` substring would be fine. This could then error out for other errors

Reply:
for the docker case, this is not a problem because we're making use of the python API, so any error there would result in an exception being thrown

Reply:
Alright got it I totally understand this.

Reply:
I think this is the only thing remaining because everything works except the result

Reply:
I figured out a better way to handle this. The error code when the object isn't available is `125`. I'll test that and if it isn't 125, raise an exception